### ISSUE TITLE
Checksum-based review

### ISSUE THREAD

**Issue by benasher44** [github/licensed#643]
It would be cool if instead of pinning reviews to a version, we could instead pin to a license checksum. This way, you can freely update the version without touching reviewed libraries, as long as the cached licensed text hasn't changed.

**Comment by benasher44** [comment]
Looks like this already kind of works via the review_changed_license flag. In that case, do I really need to use versions in my "reviewed" config?

**Comment by benasher44** [comment]
I'm using the yarn source. I think the answer is I can safely do `<package>@*` for all of my reviewed packages, since if the license changes when upgrading (even with multiple versions in play), the review_changed_license will be set, right?

**Comment by jonabc** [comment]
Yep, if the license changes when upgrading the `review_changed_license` flag should be set.  If that doesn't happen as expected let me know, that sounds like it would be a bug or other unexpected behavior that should be fixed 🙏 .

Are you getting any errors when you leave out the `@*`?  You shouldn't have to specify a version for `yarn` dependencies in the configuration file, so if you do that would also be a bug that should be fixed.

**Comment by benasher44** [comment]
Nope confirmed removing `@*` works! Just an oversight on my end.

For checksum-based review, one way to do it might be to have a global configuration that enables this "checksum mode", and then for all dependencies + matching reviewed/ignored dependencies, the cache will store the license based on its checksum rather than per dependency version. The checksum can live in the cache, where users can largely ignore it.

**Comment by benasher44** [comment]
No I think you're right. I think right now if you just don't add version requirements in you reviewed list, you basically get what I'm looking for (only pestered if license changes). Will close for now and reopen if I have a better case after using this some more. Thanks for your time!
